### PR TITLE
#297 fixed contact view

### PR DESCRIPTION
**Pull request by JoGir**

### CODE REVIEW

File: qabelbox/src/main/java/de/qabel/qabelbox/exceptions/QblStorageEntityExistsException.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package de.qabel.qabelbox.exceptions;
+
+/**
+ * Created by JoGir on 04.03.16.
```
Comment (julianseeger) [inline qabelbox/src/main/java/de/qabel/qabelbox/exceptions/QblStorageEntityExistsException.java]: please configure your IDE to not add those headers

**Comment by audax** [comment]
Next time: Just update your current PR by pushing to the same branch.

The previous point still stands:
In future PRs, please start with a new test that reproduces the problem you want to fix, even though it may be hard and annoying to do so.
It is not about you fixing a bug, it is about the next person that dives into the code. If you give that person a starting point to add new tests, it becomes much easier to work with the code. It reduces the barrier of entry for creating a proper test suite with every commit.

File: qabelbox/src/androidTest/java/de/qabel/qabelbox/services/LocalQabelServiceTest.java [hunk 2]
```diff
@@ -61,24 +62,34 @@ public void testDeleteIdentity() {
...
 	}
 
-	public void testAddContact() {
+	public void testAddContact() throws QblStorageEntityExistsException {
 		mService.addContact(contact);
 		assertTrue(mService.getContacts(identity).getContacts().contains(contact));
 	}
 
-	public void testDeleteContact() {
+	public void testAddDublicateContact() throws QblStorageEntityExistsException {
```
Comment (audax) [inline qabelbox/src/androidTest/java/de/qabel/qabelbox/services/LocalQabelServiceTest.java]: *Duplicate

File: qabelbox/src/main/java/de/qabel/qabelbox/services/LocalQabelService.java [hunk 3]
```diff
@@ -155,8 +157,15 @@ public Contacts getContacts(Identity identity) {
 		return contacts;
 	}
 
-	public void addContact(Contact contact) {
+	public void addContact(Contact contact) throws QblStorageEntityExistsException {
 		Contacts contacts = getContacts();
+		Iterator<Contact> iterator = contacts.getContacts().iterator();
```
Comment (audax) [inline qabelbox/src/main/java/de/qabel/qabelbox/services/LocalQabelService.java]: Isn't there a method like contacts.contains(contact) ?

File: qabelbox/src/main/java/de/qabel/qabelbox/activities/MainActivity.java [hunk 4]
```diff
@@ -1058,8 +1058,7 @@ private void selectManageIdentitiesFragment() {
 	private void selectContactsFragment() {
 
 		fab.show();
-		Identity activeIdentity = mService.getActiveIdentity();
-		contactFragment = ContactFragment.newInstance(mService.getContacts(activeIdentity), activeIdentity);
+		contactFragment = ContactFragment.newInstance();
```
Comment (audax) [inline qabelbox/src/main/java/de/qabel/qabelbox/activities/MainActivity.java]: new ContactFragment()

**Comment by jmt85** [comment]
@audax if everything is ok, you can close the pr

**Comment by wischweh** [comment]
will continue working on this and will commit a new PR

**Pull request closed without merging**